### PR TITLE
Script to fixup noteboks prior to committing.

### PR DESCRIPTION
**Pull request by groutr**
This script will:
  * Clear output cells
  * Clear Python metadata
  * Try to reformat code blocks with black

ping: @jameshalgren

### CODE REVIEW

File: notebooks/formatnb.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def black_code(nb):
+    fm = black.FileMode()
+    for cell in nb.cells:
+        if cell["cell_type"] == "code":
+            try:
+                cell["source"] = black.format_str(cell["source"], mode=fm)
+            except black.InvalidInput:
+                continue
```
Comment (groutr) [inline notebooks/formatnb.py:20]: Perhaps print something about possibly invalid python code encountered?

Comment (jameshalgren) [reply notebooks/formatnb.py:20]: The %cell_magic lines and !shell_execution lines trigger that exception...

Comment (jameshalgren) [reply notebooks/formatnb.py:20]: Something like this: 
```
+++ b/notebooks/formatnb.py
@@ -1,5 +1,6 @@
 import nbformat
 import black
+from sys import exit


 def clean_outputs(nb):
@@ -12,12 +13,19 @@ def clean_outputs(nb):

 def black_code(nb):
     fm = black.FileMode()
-    for cell in nb.cells:
+    for cellnum, cell in enumerate(nb.cells):
         if cell["cell_type"] == "code":
             try:
                 cell["source"] = black.format_str(cell["source"], mode=fm)
             except black.InvalidInput:
-                continue
+                print(f"Possible invalid python encountered in cell {cellnum}")
+                print(f"cell source causing exception:\n{cell['source']}\n")
+                print(r"Script will abort (leaving your file unchanged) unless you type 'continue'")
+                abort = input()
+                if abort == 'continue':
+                    continue
+                else:
+                    exit()
     return nb
```

Comment (groutr) [reply notebooks/formatnb.py:20]: Notebook magics are not valid Python. Most cell magics can be isolated to their own cell.

I also don't think we want to make the tool interactive. It shouldn't add cognitive overhead to the user.

File: notebooks/formatnb.py [hunk 2]
```diff
@@ -12,11 +12,13 @@ def clean_outputs(nb):
 
 def black_code(nb):
     fm = black.FileMode()
-    for cell in nb.cells:
+    for cellnum, cell in enumerate(nb.cells):
         if cell["cell_type"] == "code":
             try:
                 cell["source"] = black.format_str(cell["source"], mode=fm)
             except black.InvalidInput:
+                print(f"Possible invalid python encountered in cell {cellnum}")
```
Comment (groutr) [inline notebooks/formatnb.py:20]: You'll probably want `enumeate(nb.cells, 1)` to start the numbering at 1.

**Pull request merged**